provider/mesh/peer.go

```diff
@@ -0,0 +1,241 @@
+package mesh
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/alertmanager/provider"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/log"
+	"github.com/prometheus/common/model"
+	"github.com/satori/go.uuid"
+	"github.com/weaveworks/mesh"
+)
+
+type NotificationInfos struct {
```

Comment:
By that it would rather be `NotificationsInfos` – I'd love to have generally better naming for the whole area but no ideas so far.